### PR TITLE
i18n(es): update `i18n` & `plugins`

### PR DESCRIPTION
**Pull request by dreyfus92**
#### Description

- What does this PR change? Give us a brief description.
Updates `i18n` & `plugins` based on this commit 5269aad928773ae08b35ba8e19c0f2832d0d2c89

### CODE REVIEW

File: docs/src/content/docs/es/reference/plugins.md [hunk 1]
```diff
@@ -162,3 +162,71 @@ El ejemplo anterior registrará un mensaje que incluye el mensaje de informació
 ```shell
 [long-process-plugin] Empezando un proceso largo…
 ```
+
+#### `injectTranslations`
+
+**tipo:** `(translations: Record<string, Record<string, string>>) => void`
+
+Una función callback para añadir o actualizar las strings de traducción utilizadas en las [APIs de localización](/guides/i18n/#usar-traducciones-de-ui) de Starlight.
```
Comment (dreyfus92) [inline docs/src/content/docs/es/reference/plugins.md:170]:
```suggestion
Una función callback para añadir o actualizar las strings de traducción utilizadas en las [APIs de localización](/es/guides/i18n/#usar-traducciones-de-ui) de Starlight.
```

File: docs/src/content/docs/es/reference/plugins.md [hunk 2]
```diff
@@ -162,3 +162,71 @@ El ejemplo anterior registrará un mensaje que incluye el mensaje de informació
...
+        fr: {
+          'myPlugin.doThing': 'Faire le truc',
+        },
+      });
+    },
+  },
+};
+```
+
+Para usar las traducciones inyectadas en tu plugin de UI, sigue la [guía “Usar traducciones de UI”](/guides/i18n/#usar-traducciones-de-ui).
```
Comment (dreyfus92) [inline docs/src/content/docs/es/reference/plugins.md:193]:
```suggestion
Para usar las traducciones inyectadas en tu plugin de UI, sigue la [guía “Usar traducciones de UI”](/es/guides/i18n/#usar-traducciones-de-ui).
```

File: docs/src/content/docs/es/guides/i18n.mdx [hunk 3]
```diff
@@ -277,6 +276,118 @@ export const collections = {
...
+#### `t.exists()`
+
+Para comprobar si una clave de traducción existe para un idioma, utiliza la función `locals.t.exists()` con la clave de traducción como primer argumento.
+Pasa un segundo argumento opcional si necesitas anular el idioma actual.
+
+```astro
+---
+// src/components/Example.astro
+const keyExistsInCurrentLocale = Astro.locals.t.exists('a.key');
+//    ^ verdadero
```
Comment (HiDeoo) [inline docs/src/content/docs/es/guides/i18n.mdx:365]:
```suggestion
//    ^ true
```

These are boolean values returned by the function.

File: docs/src/content/docs/es/guides/i18n.mdx [hunk 4]
```diff
@@ -277,6 +276,118 @@ export const collections = {
...
+const keyExistsInFrench = Astro.locals.t.exists('another.key', { lng: 'fr' });
+//    ^ falso
```
Comment (HiDeoo) [inline docs/src/content/docs/es/guides/i18n.mdx:367]:
```suggestion
//    ^ false
```

**Review by HiDeoo — DISMISSED** [review]
Thanks for the translation :star2:

As a non-native speaker, I reviewed this PR using the [Docs Translation Review Checklist](https://hackmd.io/1KYnrpUQR0SAGTzHIUoUDg?view). I left a tiny suggestion regarding 2 return types but other than that, looks good to me.

**Comment by dreyfus92** [comment]
Thank you for the code review @HiDeoo

**Review by delucis — APPROVED** [review]
Thank you for the translations and the reviews @dreyfus92 & @HiDeoo! 💖

**Pull request merged**